transactions/demo-rust/src/lib.rs

```diff
@@ -0,0 +1,202 @@
+use crate::utils::ogmios::Ogmios;
+use crate::utils::wallet::Wallet;
+use crate::utils::{convert_plutus_data, create_tx_builder, to_redeemer};
+use cardano_serialization_lib as csl;
+use cardano_serialization_lib::address::{EnterpriseAddress, StakeCredential};
+use cardano_serialization_lib::crypto::Ed25519KeyHash;
+use cardano_serialization_lib::crypto::TransactionHash;
+use cardano_serialization_lib::output_builder::TransactionOutputBuilder;
+use cardano_serialization_lib::plutus::PlutusScript;
+use cardano_serialization_lib::tx_builder::tx_inputs_builder::{PlutusWitness, TxInputsBuilder};
+use lbf_demo_plutus_api::demo::plutus::{EqDatum, EqRedeemer};
+use plutus_ledger_api::plutus_data::IsPlutusData;
+use std::collections::BTreeMap;
+
+pub mod utils;
+
+const COINS_PER_UTXO_WORD: u64 = 34_482;
+
+/// Transaction that stores a EqDatum value at the Eq Validator.
+pub fn mk_lock_tx(
```

Comment:
The idea here is to have a pure builder part (mk_lock_tx) and an effectful part (lock_tx_build_and_submit).
The latter one takes a Wallet (capable of signing and submitting) and Ogmios (this could be similary abstracted into an Indexer).
These are currently pretty arbitrary, more like an idea for future explorations. It's likely, that some functionality of Ogmios would be taken over by the Wallet, to be in line with CIP-30 https://cips.cardano.org/cip/CIP-30/
Also, we shouldn't see raw CSL data types at this level.

Reply:
Please use the naming we use in CTL part. If you think CTL naming is bad, which i do think it is :D, change it there and align. This repo should serve as a demonstration of how equivalent things are implemented in different ecosystems.

Reply:
I'll refactor the ctl side a little bit then